components/Buttons/src/MDCFloatingButton.h

```diff
@@ -20,24 +20,45 @@
 #import "MDCButton.h"
 
 /**
- Shapes for Material Floating buttons.
+ Types of Material Floating buttons.
 
  The mini size should only be used when required for visual continuity with other elements on the
  screen.
  */
 typedef NS_ENUM(NSInteger, MDCFloatingButtonShape) {
   /**
-   A 56-point circular button surrounding a 24-point square icon or short text.
+   A 56-point circular button surrounding a 24-point or 36-point square icon or short text.
    */
   MDCFloatingButtonShapeDefault = 0,
   /**
    A 40-point circular button surrounding a 24-point square icon or short text.
    */
   MDCFloatingButtonShapeMini = 1,
+};
+
+typedef NS_ENUM(NSInteger, MDCFloatingButtonMode) {
+  /**
+   The floating button is a circle with its contents centered.
+   @c type initialization argument.
+   */
+  MDCFloatingButtonModeNormal = 0,
+
   /**
-   A 56-point circular button surrounding a 36-point square icon or short text.
+   The floating button is a "pill shape" with the image to one side of the title.
    */
-  MDCFloatingButtonShapeLargeIcon = 2
+  MDCFloatingButtonModeExtended = 1,
```

Comment:
naming-nit: Consider expanded over extended.
FAB extends our MDCButton with new functionality.  When it is expanded, it displays both the image and the title.
Naming is hard.